server/gamemodes/chess.py

```diff
@@ -158,29 +166,13 @@ def validate_pawn_move(
 
     def get_pawn_moves(self, pawn: models.Piece) -> typing.Iterator[int, int]:
         """Get all possible moves for a pawn."""
-        in_front = self.get_piece(pawn.rank + pawn.side.forwards, pawn.file)
-        if not in_front:
-            yield pawn.file, pawn.rank + pawn.side.forwards
-        if (not pawn.has_moved) and not in_front:
-            two_in_front = self.get_piece(
-                pawn.rank + pawn.side.forwards * 2, pawn.file
-            )
-            if not two_in_front:
-                yield pawn.file, pawn.rank + pawn.side.forwards * 2
-        for direction in (-1, 1):
-            file = pawn.file + direction
-            rank = pawn.rank + pawn.side.forwards
-            target = self.get_piece(rank, file)
-            if not target:
-                en_passant_pawn = self.get_piece(pawn.rank, file)
-                en_passant_valid = (
-                    en_passant_pawn and en_passant_pawn.side != pawn.side
-                    and en_passant_pawn.first_move_last_turn
-                    and pawn.rank == (4 if pawn.side == models.Side.HOME else 3)
-                )
-                if en_passant_valid:
-                    yield rank, file
-            if target and target.side != pawn.side:
+        options = ((1, 0), (2, 0), (1, -1), (1, 1))
+        for absolute_rank_delta, file_delta in options:
+            rank = pawn.rank + absolute_rank_delta * pawn.side.forwards
+            file = pawn.file + file_delta
+            if (self.on_board(rank, file)
+                    and self.validate_pawn_move(pawn, rank, file)
+                    and self.hypothetical_check(pawn.side, (pawn, rank, file))):
```

Comment:
```suggestion
            if (self.on_board(rank, file)
                    and self.validate_pawn_move(pawn, rank, file)):
```
`hypothetical_check` is already called by `possible_moves`.